nsqd/protocol_v2.go

```diff
@@ -113,6 +113,9 @@ func (p *protocolV2) IOLoop(conn net.Conn) error {
 	if client.Channel != nil {
 		client.Channel.RemoveClient(client.ID)
 	}
+	if err == io.EOF && atomic.LoadInt32(&client.State) == stateClosing {
```

Comment:
If we move this code into `readLoop` it can more explicitly ignore all errors related to the _socket_ after a `CLS` exchange.

Thoughts?